htdocs/js/rcloud_bundle.js

```diff
@@ -6420,6 +6424,16 @@ RCloud.UI.init = function() {
             range.setStartAfter($('.response')[0]);
             selection.addRange(range);
         }
+        // undo
+        if(isCmdOrCtrlAndKeyCode(90)) {
+            e.preventDefault();
+            editor.step_history_undo();
+        }
+        // redo
+        if((!ui_utils.is_a_mac() && isCmdOrCtrlAndKeyCode(89)) || (ui_utils.is_a_mac() && e.keyCode == 89 && e.metaKey && e.altKey)) {
```

Comment:
this is cmd-opt-y in the mac case, when it should be cmd-shift-z. i'm going to fix this and go ahead with the merge, since the rest looks good!